core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java

```diff
@@ -278,4 +293,125 @@ public int deleteOldVersionData(Entity.EntityType entityType, long versionRetent
   public void close() throws IOException {
     SqlSessionFactoryHelper.getInstance().close();
   }
+
+  enum JDBCBackendType {
+    H2("h2"),
+    DERBY("derby"),
+    SQLITE("sqlite"),
+    MYSQL("mysql");
+
+    private final String name;
+
+    JDBCBackendType(String name) {
+      this.name = name;
+    }
+
+    public static JDBCBackendType fromName(String name) {
+      for (JDBCBackendType backend : JDBCBackendType.values()) {
+        if (backend.name.equalsIgnoreCase(name)) {
+          return backend;
+        }
+      }
+
+      throw new IllegalArgumentException("Unknown EmbeddedJDBCBackend: " + name);
+    }
+  }
+
+  private static void initJDBCBackend(Config config) {
+
+    // Default username and password for embedded database like H2
+    String jdbcUserName = "gravitino";
+    String jdbcPassword = "gravitino";
+
+    String storagePath = getStoragePath(config);
+
+    String jdbcUrl = config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_URL);
+    String jdbcType = getJDBCType(jdbcUrl);
+
+    JDBCBackendType jdbcBackend = JDBCBackendType.fromName(jdbcType);
+
+    switch (jdbcBackend) {
+      case H2:
+        initH2Backend(config, jdbcUserName, jdbcPassword, storagePath);
+        break;
+      case DERBY:
+      case SQLITE:
+        throw new UnsupportedOperationException(jdbcPassword + " is not supported yet.");
+      case MYSQL:
+        break;
+      default:
+        throw new IllegalArgumentException("Unknown EmbeddedJDBCBackend: " + jdbcBackend);
+    }
+  }
+
+  private static String getJDBCType(String jdbcURI) {
+    if (jdbcURI.startsWith("jdbc:h2")) {
+      return "h2";
+    } else if (jdbcURI.startsWith("jdbc:derby")) {
+      return "derby";
+    } else if (jdbcURI.startsWith("jdbc:sqlite")) {
+      return "sqlite";
+    } else if (jdbcURI.startsWith("jdbc:mysql")) {
+      return "mysql";
+    } else {
+      throw new IllegalArgumentException("Unknown JDBC URI: " + jdbcURI);
+    }
+  }
+
+  private static void initH2Backend(
```

Comment:
I would suggest you have new class to handle H2 related initialization things, not doing these here. Also to see if you can refactor to avoid using switch case here.

Reply:
Done, if you have time, please take another look.